### PR TITLE
Add custom eslint rules for .only and .skip

### PR DESCRIPTION
**Pull request by rachidlamouri**
For simplicity I made the rules fail on any member call expression with .only or .skip. (for example `anything.only()` would fail). We can make them target `semantic-mocha` functions later if we want to.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
 /.nyc_output/
+/eslint-local-rules/index.js
+/eslint-local-rules/dist/
```
Comment (rachidlamouri) [inline .eslintignore:3]: I tried putting these in a `.eslintignore` within the `eslint-local-rules/` dir, but it didn't work

File: eslint-local-rules/index.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// eslint-plugin-local-rules looks for this file: https://www.npmjs.com/package/eslint-plugin-local-rules
+
+module.exports = require('./dist').default;
```
Comment (rachidlamouri) [inline eslint-local-rules/index.js:3]: - [ ] Document adding new rules

File: tests/.eslintrc.js [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
 // eslint-disable-next-line import/no-commonjs
 module.exports = {
   extends: ['../.eslintrc.js'],
   rules: {
-    'mocha/max-top-level-suites': 0,
-    'func-names': 0,
-    'mocha/no-hooks-for-single-case': 0,
+    'func-names': 'off',
+    'local-rules/semantic-mocha/no-member-only': 'error',
+    'local-rules/semantic-mocha/no-member-skip': 'error',
```
Comment (jshemas) [inline tests/.eslintrc.js:9]: Should these rules default to error?

Comment (rachidlamouri) [reply tests/.eslintrc.js:9]: This is the place where you set the severity of a rule. Rules do not have an intrinsic severity level.

File: eslint-local-rules/dist/index.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+const noMemberOnly_1 = require("./semantic-mocha/noMemberOnly");
+const noMemberSkip_1 = require("./semantic-mocha/noMemberSkip");
+// eslint-disable-next-line import/no-default-export
+exports.default = {
+    [noMemberOnly_1.semanticMochaNoMemberOnly.name]: noMemberOnly_1.semanticMochaNoMemberOnly.module,
+    [noMemberSkip_1.semanticMochaNoMemberSkip.name]: noMemberSkip_1.semanticMochaNoMemberSkip.module,
+};
```
Comment (rachidlamouri) [inline eslint-local-rules/dist/index.js:9]: Update to LF instead of CRLF

File: eslint-local-rules/dist/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+declare const _default: {
```
Comment (rachidlamouri) [inline eslint-local-rules/dist/index.d.ts:1]: I don't think we need these declaration files

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -17,6 +17,9 @@ jobs:
           node-version: '16.x'
       - name: Install
         run: npm ci
+      - name: Test Custom Eslint
+        if: always()
+        run: npm run test:custom-eslint-rules
```
Comment (jshemas) [inline .github/workflows/ci.yaml:22]: Should we run `npm run compile:custom-eslint-rules` before this?

File: .eslintignore (hunk 1 above)
Comment (rachidlamouri) [reply .eslintignore]: Fixed it

**Review by jshemas — APPROVED** [review]

**Pull request merged**